bin/varnishd/cache/cache_cli.c

```diff
@@ -76,7 +76,9 @@ cli_cb_before(const struct cli *cli, struct cli_proto *clp, const char * const *
 	ASSERT_CLI();
 	(void)av;
 
-	if (clp != NULL &&  (clp->desc->flags & CLI_F_SENSITIVE)) {
+	if (clp != NULL &&
+	    !DO_DEBUG(DBG_CLI_SHOW_SENSIITIVE) &&
+	    (clp->desc->flags & CLI_F_SENSITIVE)) {
```

Comment:
We could imagine a utility function in libvarnish:

```c
unsigned
VCLS_IsSensitive(struct cli_proto *clp, unsigned show)
{

        if (clp == NULL || show)
                return (0);
        return (clp->desc->flags & CLI_F_SENSITIVE);
}
```

And then this if statement and 3 others can be revisited:

```c
if (VCLS_IsSensitive(clp, [MGT_]DO_DEBUG(DBG_CLI_SHOW_SENSITIVE)) {
        /* do the logfunc thing */
}
```

Reply:
Added in latest revision